database/database.go

```diff
@@ -140,41 +135,42 @@ func (db *DB) UserByID(ctx context.Context, id string) (*User, error) {
 	return &u, nil
 }
 
-// UserCreate creates a new user in the DB. We need the user object to be passed
-// by reference because we need to be able to update the ID of new user.
-func (db *DB) UserCreate(ctx context.Context, u *User) error {
-	if !u.Email.Validate() {
-		return ErrInvalidEmail
-	}
-	// Check for an existing user with this email.
-	users, err := db.managedUsersByField(ctx, "email", string(u.Email))
+// UserCreate creates a new user in the DB.
+func (db *DB) UserCreate(ctx context.Context, sub string, tier int) (*User, error) {
+	// Check for an existing user with this sub.
+	users, err := db.managedUsersByField(ctx, "sub", sub)
 	if err != nil && !errors.Contains(err, ErrUserNotFound) {
-		return errors.AddContext(err, "failed to query DB")
+		return nil, errors.AddContext(err, "failed to query DB")
 	}
 	if len(users) > 0 {
-		return ErrEmailAlreadyUsed
+		return nil, ErrUserAlreadyExists
+	}
+	u := &User{
+		ID:   primitive.ObjectID{},
+		Sub:  sub,
+		Tier: tier,
 	}
 	// Insert the user.
 	fields, err := bson.Marshal(u)
 	if err != nil {
-		return err
+		return nil, err
 	}
 	ir, err := db.staticUsers.InsertOne(ctx, fields)
 	if err != nil {
-		return errors.AddContext(err, "failed to Insert")
+		return nil, errors.AddContext(err, "failed to Insert")
 	}
 	u.ID = ir.InsertedID.(primitive.ObjectID)
 	// Sanity check because races exist.
-	users, err = db.managedUsersByField(ctx, "email", string(u.Email))
+	users, err = db.managedUsersByField(ctx, "sub", sub)
```

Comment:
You should make the `sub` field unique, this way you'd get a `UniqueConstraintViolationException` on the second insert. It's perfectly possible to be 100% atomic in mongo, they even have a new feature called `Transactions` that allow you to be ACID across collections!

Reply:
Same comment as earlier. But I don't mind F/U, but we really have to F/U. 
The follow up is literally less than 20 lines of code by the way.

Reply:
F/U: https://github.com/NebulousLabs/skynet-accounts/issues/18